components/Header.tsx

```diff
@@ -1,11 +1,105 @@
 import Link from "next/link";
+import Image from "next/image";
+import { useRouter } from 'next/router';
+import { useEffect, useState } from "react";
 
 const Header = () => {
+    const router = useRouter();
+    const currentPath = router.pathname;
+
+    const isActive = (path: string): string => {
+        return currentPath === path ? 'border-project-green bg-project-green font-bold' : 'group';
+    };
+
+    const [toggleMenu, setToggleMenu] = useState(false);
+
+    useEffect(() => {
+        const handleResize = () => {
+        //   setWindowWidth(window.innerWidth);
+          if (window.innerWidth > 1023) {
+            setToggleMenu(false)
+          }
+        };
+    
+        // Set the initial window width
+        // setWindowWidth(window.innerWidth);
+    
+        // Attach event listener for window resize
+        window.addEventListener('resize', handleResize);
+    
+        // Clean up the event listener on component unmount
+        return () => {
+          window.removeEventListener('resize', handleResize);
+        };
+      }, []);
+
+    function setMenuState ():any {
+        setToggleMenu(() => !toggleMenu);
+        if (toggleMenu) {
+            document.body.style.overflow = "auto";
+          } else {
+            document.body.style.overflow = "hidden";
+          }
+    }
+
+    const [isScrolled, setIsScrolled] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollPosition = window.scrollY;
+      if (scrollPosition > 0) {
+        setIsScrolled(true);
+      } else {
+        setIsScrolled(false);
+      }
+    };
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
     return ( 
-        <header>
-            <h2>This is the Header all-mighty</h2>
-            <Link href="/">Home</Link>
-            <Link href="/works">Works</Link>
+        <header className="fixed top-0 w-full h-0">
+            <div className={`flex justify-between items-center px-5 md:px-14 h-[5.5rem] z-10 ease-in duration-200 ${(currentPath === "/" && !toggleMenu) ? (isScrolled && 'bg-project-black'): ('bg-project-black')}`}>
+                <Link onClick={setMenuState} href="/"><img src="/logo-white.png" className=" h-10 w-fit" alt="logo"></img></Link>
+                <nav className=" hidden lg:block">
+                    <ul className="flex gap-x-7 uppercase text-base items-center">
+                        <li className="relative"><Link href="/" className={`block px-4 py-3 ${isActive('/')}`}>Home<span className="absolute left-1/2 bottom-0 border-b-2 w-0 border-project-green transition-all duration-300 ease-in-out group-hover:w-full transform -translate-x-1/2"></span></Link></li>
```

Comment:
Please make this `li` into a component. Ideally use the same component for links used for desktop and mobile  (see line 89), too. You can pass a prop that will determine which styling will be applied